src/Snapshot.php

```diff
@@ -61,6 +61,103 @@ public function load(string $connectionName = null)
         event(new LoadedSnapshot($this));
     }
 
+    public function loadStream(string $connectionName = null)
+    {
+        event(new LoadingSnapshot($this));
+
+        if ($connectionName !== null) {
+            DB::setDefaultConnection($connectionName);
+        }
+
+        $this->dropAllCurrentTables();
+
+        if ($this->compressionExtension === 'gz') {
+            $dumpFilePath = $this->decompress();
+        } else {
+            $dumpFilePath = $this->disk->path($this->fileName);
+        }
+
+        $this->streamFileIntoDB($dumpFilePath, $connectionName);
+
+        event(new LoadedSnapshot($this));
+    }
+
+    public function streamFileIntoDB($path, string $connectionName = null)
+    {
+        if ($connectionName !== null) {
+            DB::setDefaultConnection($connectionName);
+        }
+
+        $tmpLine = '';
+        $lines = file($path);
+        $errors = [];
+
+        foreach ($lines as $line) {
+
+            // Skip it if line is a comment
+            if (substr($line, 0, 2) == '--' || trim($line) == '') {
```

Comment:
I suggest changing to `Str::startsWith($line, '--') || trim($line) == ''` for better readability